From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Guillaume <guillaume.thouvenin@vates.tech>
Date: Mon, 31 Mar 2025 11:05:15 +0200
Subject: [PATCH] configure xapi.conf to meet our needs

This patch enables:
  - custom uefi certificates
  - our additional SMAPI plugins drivers
  - creation of an SR for tools ISO
  - modification of host scheduler

It disables website https-only

Signed-off-by: Guillaume <guillaume.thouvenin@vates.tech>

diff --git a/scripts/xapi.conf b/scripts/xapi.conf
index accc1aba8088..c8e451d37703 100644
--- a/scripts/xapi.conf
+++ b/scripts/xapi.conf
@@ -1,3 +1,7 @@
+# WARNING: any changes made to this file may be overwritten by future
+# updates of the xapi-core RPM.
+# You may want to create a file in the /etc/xapi.conf.d/ directory instead
+
 # The xapi config file has 3 sections: a global policy section;
 # a section for paths to helper utilities; and a timeout tweak section.
 
@@ -91,7 +95,7 @@ igd-passthru-vendor-whitelist = 8086
 # repository-domain-name-allowlist =
 
 # Allow the use of custom UEFI certificates
-# allow-custom-uefi-certs = false
+allow-custom-uefi-certs = true
 
 # Paths to utilities: ############################################
 
@@ -173,7 +177,7 @@ sparse_dd = /usr/libexec/xapi/sparse_dd
 # sm-dir =  @OPTDIR@/sm
 
 # Whitelist of SM plugins
-sm-plugins=ext nfs iscsi lvmoiscsi dummy file hba rawhba udev iso lvm lvmohba lvmofcoe shm smb
+sm-plugins=ext nfs iscsi lvmoiscsi dummy file hba rawhba udev iso lvm lvmohba lvmofcoe shm smb cephfs glusterfs xfs zfs largeblock
 
 # Directory containing tools ISO
 # tools-sr-dir = @OPTDIR@/packages/iso
@@ -406,3 +410,12 @@ xen_livepatch_list = "/usr/sbin/xen-livepatch list"
 
 # The command to query current kernel patch list
 kpatch_list = "/usr/sbin/kpatch list"
+
+# true = forbid HTTP requests to website (403); false = allow HTTP to HTTPS redirection
+website-https-only = false
+
+# Indicates whether to create an SR for Tools ISOs
+create-tools-sr = true
+
+# Allows to modify the host's scheduler granularity
+allow-host-sched-gran-modification = true
